spec.html

```diff
@@ -115,11 +115,10 @@ <h1>Array.prototype.with ( _index_, _value_ )</h1>
                 <emu-alg>
                     1. Let _O_ be the *this* value.
                     1. Let _len_ be ? LengthOfArrayLike(_O_).
-                    1. If IsIntegralNumber(_index_) is *false*, throw a *RangeError* exception.
-                    1. If _index_ &ge; _len_, throw a *RangeError* exception.
-                    1. If _index_ &lt; 0, let _actualIndex_ be _len_ + _index_.
-                    1. Else, let _actualIndex_ be _index_.
-                    1. If _actualIndex_ &lt; 0, throw a *RangeError* exception.
+                    1. Let _relativeIndex_ be ? ToIntegerOrInfinity(_index_).

```

Comment:
this changes the error for a BigInt from a RangeError to a TypeError. It also allows `1.1` where before it would throw. Is that intentional?